### PR TITLE
Always check if upgrading to the allowed RHEL version

### PR DESCRIPTION
**Pull request by bocekm**
Running the Preupgrade Assistant is a mandatory prerequisite before upgrading the system with the Red Hat Upgrade Tool. The Preupgrade Assistant performs a set of checks and many of them are relevant for one specific upgrade "path" only, e.g. for upgrades of RHEL 6.9 to RHEL 7.4. Because the Red Hat Upgrade Tool works with the information provided by the Preupgrade Assistant, the Red Hat Upgrade Tool needs to allow upgrading the system **only** to such a RHEL version for which the Preupgrade Assistant analysis has been designed for.

There was a limited[1] check just for _network_ as a source; for _iso_ or _device_ sources there wasn't any check at all. Now, the RHEL version specified in the _.treeinfo_ file of the installation media (be it network, iso or device) need to match the RHEL version specified by the Preupgrade Assistant.

This PR also includes:
- rewording some of the user-facing messages
- code cleanup: spaghetti into smaller functions

[1] The limited check was comparing the RHEL version passed through the `--network` option against the Preupgrade Assistant-generated data.

Resolves: https://github.com/upgrades-migrations/redhat-upgrade-tool/issues/30

### CODE REVIEW

**Comment by pirat89** [comment]
Code seems OK, I will test it during monday.

File: redhat-upgrade-tool.py
Comment (jhornice) [inline redhat-upgrade-tool.py:139]: the Preupgrade Assistant

File: redhat-upgrade-tool.py
Comment (jhornice) [inline redhat-upgrade-tool.py:181]: Skipping the system variant check.

File: redhat-upgrade-tool.py
Comment (jhornice) [inline redhat-upgrade-tool.py:261]: I do not understand

Comment (bocekm) [reply redhat-upgrade-tool.py:261]: This change (space after # and start with capital letter) is really just formal and was not necessary. But I did it:
- to follow PEP formating (https://www.python.org/dev/peps/pep-0008/#block-comments)
- to have the comment consistent with the other comments

**Review by artmello — COMMENTED** [review]
code lgtm

File: redhat_upgrade_tool/commandline.py
Comment (pirat89) [inline redhat_upgrade_tool/commandline.py]: Should be *ampersand* really here ?

**Comment by pirat89** [comment]
Just waiting for answer before approval (the solutions works as expected).

File: redhat_upgrade_tool/commandline.py
Comment (bocekm) [reply redhat_upgrade_tool/commandline.py]: Correct, it should be dollar sign. Thanks.

**Comment by bocekm** [comment]
@jhornice, please check the English of the latest commit changes. Thanks.

**Review by pirat89 — DISMISSED** [review]
Giving approve, so it can be merged after the verification by @jhornice when there it is not error :-)

**Comment by pkotvan** [comment]
This is pretty easy to reproduce and verify. Just need to try to upgrade to some older minor version of RHEL-7. Ideally using at least a network source and iso.

**Comment by pkotvan** [comment]
When I tried to verify this I encountered this traceback when calling `redhat-upgrade-tool`:

```
# redhat-upgrade-tool                        
Traceback (most recent call last):                                  
  File "/usr/bin/redhat-upgrade-tool", line 35, in <module>         
    from redhat_upgrade_tool.commandline import parse_args, do_cleanup, device_setup                                                    
  File "/usr/lib/python2.6/site-packages/redhat_upgrade_tool/commandline.py", line 238, in <module>                                     
    class Option(optparse.Option):                                  
  File "/usr/lib/python2.6/site-packages/redhat_upgrade_tool/commandline.py", line 244, in Option                                       
    TYPE_CHECKER["VERSION"] = VERSION                               
NameError: name 'VERSION' is not defined
```

I'm not sure if this is me doing something wrong, or this is caused by this change.

**Comment by pkotvan** [comment]
I was able to verify the fix. Upgrade attempt from RHEL-6.9 to RHEL-7.3 using `--iso` and `--network` both exitting with following message *The installed version of Preupgrade Assistant allows upgrade only to the system version 7.4.*

**Review by pirat89 — APPROVED** [review]

**Pull request merged**
